src/util/TypeName.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <string_view>
+
+namespace chatterino {
+
+// Adapted from: https://stackoverflow.com/a/56766138.
+// NOTE: Relies on the "magic" prefixes and suffixes. There are implementations
+// that attempt to manually detect these (linked in the SO answer above) but
+// they seemed too complex for the scope we have here.
+template <typename T>
+constexpr auto type_name()
```

Comment:
I think we can use typeid(t).name() here? 
https://en.cppreference.com/w/cpp/language/typeid

Reply:
https://godbolt.org/z/W65sv4deo